common/main/java/com/couchbase/lite/internal/utils/Preconditions.java

```diff
@@ -75,6 +75,13 @@ public static int assertNotZero(int n, @NonNull String name) {
         return n;
     }
 
+    public static long assertUnsigned(long n, @NonNull String name) {
+        if ((n < 0) || (n >= (1L << 33))) {
```

Comment:
I'm guessing this is a range check for 32-bit uint?  If so it needs to check >= 1L << 32, not 33.  The max value of uint is 2^32 - 1